crates/module-system/sov-solana-offchain-auth/src/capabilities.rs

```diff
@@ -0,0 +1,170 @@
+use std::marker::PhantomData;
+
+use borsh::{BorshDeserialize, BorshSerialize};
+use serde::de::DeserializeOwned;
+use serde::Serialize;
+use sov_modules_api::capabilities::{
+    AuthenticationError, BatchFromUnregisteredSequencer, FatalError, TransactionAuthenticator,
+    UnregisteredAuthenticationError,
+};
+use sov_modules_api::{DispatchCall, FullyBakedTx, ProvableStateReader, RawTx, Runtime, Spec};
+
+/// Indicates that a runtime supports the `SolanaOffchain` transaction authenticator
+/// and provides suitable methods for encoding and decoding solana offchain message transactions.
+pub trait SolanaOffchainAuthenticatorTrait<S: Spec>: Runtime<S> {
```

Comment:
At some point, we discussed the idea of unifying all authenticators (though I may have misunderstood). Is there a way to achieve this functionality without implementing these additional traits?